static_frame/core/index_hierarchy.py

```diff
@@ -1051,14 +1050,65 @@ def gen() -> tp.Iterator[tp.Tuple[tp.Hashable, ...]]:
                 )
 
 
-    # def relabel_at_depth(self: IH,
-    #         mapper: RelabelInput,
-    #         depth_level: DepthLevelSpecifier = 0
-    #         ) -> IH:
-    #     '''
-    #     Return a new :obj:`IndexHierarchy` after applying the hte `mapper` to the depth level or levels specified by `depth_level`.
-    #     '''
-    #     pass
+    def relabel_at_depth(self,
+            mapper: RelabelInput,
+            depth_level: DepthLevelSpecifier = 0
+            ) -> "IndexHierarchy":
+        '''
+        Return a new :obj:`IndexHierarchy` after applying the the `mapper` to the depth level or levels specified by `depth_level`.
+        '''
+        if self._recache:
+            self._update_array_cache()
+
+        if isinstance(depth_level, INT_TYPES):
+            # Single depth levels have an optimized path through IndexLevels
+            return self.__class__(
+                    levels=self._levels.relabel_at_specific_depth(mapper, depth_level),
+                    name=self._name,
+                    )
+
+        depth_level = list(depth_level)
+
+        if len(depth_level) == 1:
+            # Optimized path for single depth level
+            return self.__class__(
+                levels=self._levels.relabel_at_specific_depth(mapper, depth_level[0]),
+                name=self._name,
+                )
+        elif len(depth_level) == self.depth:
+            # Equivalent to relabel!
+            return self.relabel(mapper)
+
+        index_constructors = tuple(self._levels.index_types())
+
+        def get_modified(array_items: tuple) -> tp.Iterable[tp.Any]: # type: ignore
+            if callable(mapper):
+                return mapper(array_items) # type: ignore
+
+            # if a mapper, it must support both __getitem__ and __contains__
+            return mapper.get(array_items, array_items) # type: ignore
```

Comment:
I think our expectation is that, if multiple depth levels are provided, the same mapper will be applied on each element in each depth. It looks here that we are mapping/processing the entire row, which is plausible interpretation of what this function does, but maybe not something we need to support.

Reply:
Ah okay that's what @nekobon said she expected too. Although my interpretation of the commented out stub was row-based. If that's incorrect, then I will update it. Are you wanting the other implementation? I'm not exactly sure what that would mean though. For example, how would you specify a relabel at multiple levels using a dictionary? Would you provide a sequence of dicts?

```python
ih.relabel_at_depth([dict(A=1), dict(I="i")), levels=[0,1])
```
?